keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -361,6 +355,57 @@ def next(prompt, cache, index):
             hidden_states=hidden_states,
         )
 
+        # Compute an output padding mask with the token ids we updated.
+        if end_token_id is not None:
+            end_locations = (token_ids == end_token_id) & (~padding_mask)
+            end_locations = tf.cast(end_locations, tf.int32)
+            overflow = tf.math.cumsum(end_locations, exclusive=True)
```

Comment:
Maybe worth a comment saying all positions after first end_token in overflow will have value > 1.